resource-aggregate/pb/events.proto

```diff
@@ -16,69 +16,72 @@ message EventMetadata {
-    string device_id = 2;
-    AuditContext audit_context = 3;
-    EventMetadata event_metadata = 4;
+    string device_id = 1;
+    repeated Resource resources = 2;
+    AuditContext audit_context = 100;
+    EventMetadata event_metadata = 101;
 }
 
 // https://github.com/openconnectivityfoundation/cloud-services/blob/master/swagger2.0/oic.wk.rd.swagger.json #Specification CR needed
 message ResourceLinksUnpublished {
-    repeated string hrefs = 1;
-    string device_id = 2;
-    AuditContext audit_context = 3;
-    EventMetadata event_metadata = 4;
+    string device_id = 1;
+    repeated string hrefs = 2;
+    AuditContext audit_context = 100;
+    EventMetadata event_metadata = 101;
 }
 
 message ResourceLinksSnapshotTaken {
-    map<string, Resource> resources = 1;
-    string device_id = 2;
-    EventMetadata event_metadata = 3;
+    string device_id = 1;
+    map<string, Resource> resources = 2;
+    EventMetadata event_metadata = 101;
 }
 
 message ResourceChanged {
     ResourceId resource_id = 1;
     Content content = 2;
-    AuditContext audit_context = 3;
-    EventMetadata event_metadata = 4;
     Status status = 5;
+    AuditContext audit_context = 100;
+    EventMetadata event_metadata = 101;
 }
 
 message ResourceUpdatePending {
     ResourceId resource_id = 1;
     string resource_interface = 2;
     Content content = 4;
-    AuditContext audit_context = 5;
-    EventMetadata event_metadata = 6;
+    AuditContext audit_context = 100;
+    EventMetadata event_metadata = 101;
+    int64 valid_until = 102;  // unix timestamp in nanoseconds (https://golang.org/pkg/time/#Time.UnixNano) when pending event is considered as expired. 0 means forever.
 }
 
 message ResourceUpdated {
     ResourceId resource_id = 1;
     Status status = 2;
     Content content = 3;
-    AuditContext audit_context = 4;
-    EventMetadata event_metadata = 5;
+    AuditContext audit_context = 100;
+    EventMetadata event_metadata = 101;
 }
 
 message ResourceRetrievePending {
     ResourceId resource_id = 1;
     string resource_interface = 2;
-    AuditContext audit_context = 3;
-    EventMetadata event_metadata = 4;
+    AuditContext audit_context = 100;
+    EventMetadata event_metadata = 101;
+    int64 valid_until = 102;  // unix timestamp in nanoseconds (https://golang.org/pkg/time/#Time.UnixNano) when pending event is considered as expired. 0 means forever.
 }
 
 message ResourceRetrieved {
     ResourceId resource_id = 1;
     Status status = 2;
     Content content = 3;
-    AuditContext audit_context = 4;
-    EventMetadata event_metadata = 5;
+    AuditContext audit_context = 100;
+    EventMetadata event_metadata = 101;
 }
 
 message ResourceDeletePending {
     ResourceId resource_id = 1;
     AuditContext audit_context = 100;
     EventMetadata event_metadata = 101;
+    int64 valid_until = 102;  // unix timestamp in nanoseconds (https://golang.org/pkg/time/#Time.UnixNano) when pending event is considered as expired. 0 means forever.
```

Comment:
field numbers in the range 1 through 15 take one byte to encode, while 15+ takes 2 bytes. 1-15 shall be used  for **very frequently occurring message elements**. validUntil, auditContext and eventMetadata are always present and shall be imho always in range 1-15.

Reply:
2bytes is not big deal for sharing id.

Reply:
But why to not have one. There is extremely low chance that we will occupy additional 15-1+3 1byte positions.

Reply:
Fixed